src/lib.rs

```diff
@@ -126,23 +127,69 @@ impl Debug for GetResult {
 impl GetResult {
     /// Collects the data into a [`Bytes`]
     pub async fn bytes(self) -> Result<Bytes> {
-        collect_bytes(self.into_stream(), None).await
+        match self {
+            Self::File(mut file, path) => {
+                maybe_spawn_blocking(move || {
```

Comment:
I think it is worth a comment here explaining the rationale for special casing `Self::File` (basically copy/paste the description from the PR that this is an optimization to avoid tokio::fs overhead)